packages/react-sdk/src/components/elements/image/ImageDisplay.tsx

```diff
@@ -64,132 +157,21 @@ function ImageDisplay({
   overrides = {},
 }: ImageDisplayProps) {
   const widgetApi = useWidgetApi();
-  const [loadError, setLoadError] = useState(false);
-  const [loading, setLoading] = useState(true);
-  const [imageUri, setImageUri] = useState<undefined | string>();
+  const { data: imageUri } = useSWR(
+    { widgetApi, baseUrl, mxc, mimeType },
+    downloadFile,
```

Comment:
We cant call downloadFile directly as we want values from props. It would rerender contantly. SWR glues this (and also gives us caching)